File: benchmarks/reference/primes.py

import time

def is_prime(n):
    if n <= 1: return False
    i = 2
    while i * i <= n:
        if n % i == 0: return False
        i += 1
    return True

if __name__ == "__main__":
    start = time.time()
    count = 0
    for i in range(5000):
        if is_prime(i):
            count += 1
    print(f"Primes up to 5000: {count}")
    elapsed = time.time() - start
    print(f"Time: {elapsed:.6f}")

File: benchmarks/reference/primes.cpp
#include <iostream>
#include <chrono>

bool is_prime(int n) {
    if(n <= 1) return false;
    for(int i=2; i*i<=n; i++) {
        if(n % i == 0) return false;
    }
    return true;
}

int main() {
    auto start = std::chrono::high_resolution_clock::now();
    int count = 0;
    for(int i=0; i<5000; i++) {
        if(is_prime(i)) count++;
    }
    std::cout << "Primes up to 5000: " << count << "\n";
    std::chrono::duration<double> elapsed = std::chrono::high_resolution_clock::now() - start;
    std::cout << "Time: " << elapsed.count() << "\n";
    return 0;
}
